apps/vth-frontend/src/util/getNavData.ts

```diff
@@ -0,0 +1,60 @@
+'use server';
+
+import { NavigationListType } from '@frameless/ui';
+import { GET_NAVIGATION_DATA } from '@/query';
+import { createStrapiURL } from './createStrapiURL';
+import { fetchData } from './fetchData';
+
+type NavAttributes = {
+  attributes: {
+    title: string;
+    slug: string;
+    theme_pages?: { data: NavAttributes[] };
+    article_pages?: { data: NavAttributes[] };
+  };
+};
+
+interface NavTypes {
+  data: { navigationPages: { data: NavAttributes[] } };
+}
+
+interface GetNavDataType {
+  pageMode?: string;
+}
+
+export const getNavData = async ({ pageMode }: GetNavDataType) => {
+  const { data }: NavTypes = await fetchData({
+    url: createStrapiURL(),
+    query: GET_NAVIGATION_DATA,
+    variables: { pageMode },
+  });
+  const isNavList = Array.isArray(data?.navigationPages?.data);
+  const navListData =
+    isNavList &&
+    (data.navigationPages.data.map((navigationPage) => {
+      const isThemeList = Array.isArray(navigationPage.attributes.theme_pages?.data);
+      const isArticleList = Array.isArray(navigationPage.attributes.article_pages?.data);
+
+      const themeList =
+        isThemeList &&
+        navigationPage.attributes.theme_pages?.data.map(({ attributes: { slug, title } }) => ({
+          textContent: title,
+          href: `/theme/${slug}`,
+        }));
+
+      const articleList =
+        isArticleList &&
+        navigationPage.attributes.article_pages?.data.map(({ attributes: { slug, title } }) => ({
+          textContent: title,
+          href: `/article/${slug}`,
+        }));
+
+      return {
+        textContent: navigationPage.attributes.title,
+        href: `/${navigationPage.attributes.slug}`,
+        children: themeList || articleList,
```

Comment:
Ik denk dat we deze samen horen te voegen! Soms zit er namelijk een artikel direct onder een navigatie pagina. Kijk bijvoorbeeld bij [Bestaande gebouwen](https://vth-programma-verslag.digitaal.utrecht.nl/nl/bestaande-gebouwen) -> [Meldingen en controles](Meldingen en controles gebouwen)